consensus/istanbul/core/roundchange.go

```diff
@@ -126,18 +126,23 @@ func (c *core) handleRoundChangeCertificate(proposal istanbul.Subject, roundChan
 		if err := message.Decode(&roundChange); err != nil {
 			logger.Warn("Failed to decode ROUND CHANGE in certificate", "err", err)
 			return err
+		} else if roundChange.View == nil || roundChange.View.Sequence == nil || roundChange.View.Round == nil {
+			return errInvalidRoundChangeCertificateMsgView
 		}
-		msg_logger := logger.New("msg_round", roundChange.View.Round, "msg_seq", roundChange.View.Sequence)
 
-		// Verify ROUND CHANGE message is for a proper view
-		if roundChange.View.Cmp(proposal.View) != 0 || roundChange.View.Round.Cmp(c.current.DesiredRound()) < 0 {
-			msg_logger.Error("Round change in certificate for wrong view", "err", err)
+		msgLogger := logger.New("msg_round", roundChange.View.Round, "msg_seq", roundChange.View.Sequence)
+
+		// Verify ROUND CHANGE message is for the same sequence AND an equal or subsequent round as the proposal.
+		// We have already called checkMessage by this point and checked the proposal's and PREPREPARE's sequence match.
+		if roundChange.View.Sequence.Cmp(proposal.View.Sequence) != 0 || roundChange.View.Round.Cmp(proposal.View.Round) < 0 {
+			msgLogger.Error("Round change in certificate for an earlier view", "roundChange.View", "err", err)
```

Comment:
RoundChange view not being logged here. Also nit, but technically round change message could be for a future view if it's a future sequence